### PR TITLE
feat(Modal): sync modal styles with pattern library and include alignment props

### LINKED ISSUE #990: Sync Modal with UXPin
Update the dismiss action to match the new design:

modal with close action

`<ModalContent>` and `<ModalFooter>` should allow for alignment, i.e. `align="right"`. Defaults as `align="left"`.

Update the padding to the `ModalFooter` to match `16px` all around.

### PR DESCRIPTION
**Pull request by scurker**
closes #990

### CODE REVIEW

File: packages/styles/dialog.css [hunk 1]
```diff
@@ -166,20 +177,14 @@
...
+  color: var(--accent-medium);
 }
 
 .cauldron--theme-dark .Modal--info .Dialog__header,
 .cauldron--theme-dark .Modal--info .Dialog__content,
 .cauldron--theme-dark .Modal--info .Dialog__footer {
-  background-color: var(--accent-medium);
-}
-
-.cauldron--theme-dark .Modal--info .Dialog__header {
```
Comment (dequejosie) [inline packages/styles/dialog.css:181]: I wonder if it is necessary to add the border-bottom for info Modal. At least the new ad modal I am looking at does not have that.

**Comment by dequejosie** [comment]
Left one question. Not blocking. Changes make sense to me.

**Review by dequejosie — APPROVED** [review]

File: packages/styles/dialog.css (hunk 1 above)
Comment (scurker) [reply packages/styles/dialog.css:181]: This is what's in the pattern library at least for the "simple" variation: https://preview.uxpin.com/520e80315786a67ba43a3ca50598e6d69782627c#/pages/161871161/simulate/no-panels?mode=i

Comment (dequejosie) [reply packages/styles/dialog.css:181]: I see. I guess we will have to remove the bottom border for multiple places in the extension.

**Pull request merged**